mobilizon_reshare/cli/commands/publish/main.py

```diff
@@ -1,10 +1,39 @@
-from mobilizon_reshare.main.publish import publish
+import logging
 
+from mobilizon_reshare.main.publish import (
+    select_and_publish,
+    publish_event,
+    publish_publications,
+)
+from mobilizon_reshare.publishers import get_active_publishers
+from mobilizon_reshare.storage.query.read import get_mobilizon_event, get_publication
 
-async def publish_command():
+logger = logging.getLogger(__name__)
+
+
+async def publish_command(event, publication, platform, list_platforms):
     """
-    STUB
-    :return:
+    If no arguments are provided, select an event with the current configured strategy
+    and publish it to all active platforms.
+
+    - If list_platforms is True print all platforms and exit
+    - Otherwise:
+       + If publication is not None, publish it.
+       + Otherwise if event is not None publish it.
     """
-    reports = await publish()
-    return 0 if reports and reports.successful else 1
+    if publication and event:
+        logger.error("-P/--publication and -E/--event are incompatible!")
```

Comment:
this should be handled at click level, not here

Reply:
Fixed, thanks.